--- libquirc-1.0.1/Makefile.orig	2024-04-13 15:39:22.221266168 +0200
+++ libquirc-1.0.1/Makefile	2024-04-13 15:40:51.247450539 +0200
@@ -15,8 +15,8 @@
 
 CC ?= gcc
 PREFIX ?= /usr/local
-SDL_CFLAGS ?= pkg-config --cflags sdl 
-SDL_LIBS ?= pkg-config --libs sdl
+SDL_CFLAGS ?= $$(pkg-config --cflags sdl)
+SDL_LIBS ?= $$(pkg-config --libs sdl)
 
 LIB_VERSION = 1.0
 
@@ -57,7 +57,7 @@ libquirc.a: $(LIB_OBJ)
 libquirc.so: libquirc.so.$(LIB_VERSION)
 
 libquirc.so.$(LIB_VERSION): $(LIB_OBJ)
-	$(CC) -shared -o $@ $(LIB_OBJ) $(LDFLAGS) -lm
+	$(CC) -shared -Wl,-soname,libquirc.so.$(LIB_VERSION) -o $@ $(LIB_OBJ) $(LDFLAGS) -lm
 
 .c.o:
 	$(CC) $(QUIRC_CFLAGS) -o $@ -c $<
